src/lib/markbind/src/preprocessors/variablePreprocessor.js

```diff
@@ -0,0 +1,468 @@
+    const importedVariables = {};
+    $('import[from]').each((index, element) => {
+      const variableNames = Object.keys(element.attribs).filter(name => name !== 'from' && name !== 'as');
+
+      // Add pseudo variables for the alias's variables
+      const alias = VariablePreprocessor.getAliasForImportVariables(variableNames, element);
+      importedVariables[alias] = new Proxy({}, {
+        get(obj, prop) {
+          return `{{${alias}.${prop}}}`;
+        },
+      });
+
+      // Add pseudo variables for inline variables in the import element
+      variableNames.forEach((name) => {
+        importedVariables[name] = `{{${alias}.${name}}}`;
+      });
+    });
+    return importedVariables;
+  }
+
+  /**
+   * Extract page variables from a page.
+   * @param fileName for error printing
+   * @param data to extract variables from
+   * @param includedVariables variables from parent include
+   */
+  extractPageVariables(fileName, data, includedVariables) {
+    const fileDir = path.dirname(fileName);
+    const $ = cheerio.load(data);
+
+    const importedVariables = VariablePreprocessor.extractPseudoImportedVariables($);
+    const pageVariables = {};
+    const userDefinedVariables = this.getContentParentSiteVariables(fileName);
+
+    const setPageVariableIfNotSetBefore = (variableName, rawVariableValue) => {
+      if (pageVariables[variableName]) {
+        return;
+      }
+
+      pageVariables[variableName] = njUtil.renderRaw(rawVariableValue, {
+        ...importedVariables,
+        ...pageVariables,
+        ...userDefinedVariables,
+        ...includedVariables,
+      }, {}, false);
+    };
+
+    $('variable').each((index, node) => {
+      const variableSource = $(node).attr('from');
+      if (variableSource !== undefined) {
+        try {
+          const variableFilePath = path.resolve(fileDir, variableSource);
+          const jsonData = fs.readFileSync(variableFilePath);
+          const varData = JSON.parse(jsonData);
+          Object.entries(varData).forEach(([varName, varValue]) => {
+            setPageVariableIfNotSetBefore(varName, varValue);
+          });
+        } catch (err) {
+          console.warn(`Error ${err.message}`);
+        }
+      } else {
+        const variableName = $(node).attr('name');
+        if (!variableName) {
+          console.warn(`Missing 'name' for variable in ${fileName}\n`);
+          return;
+        }
+        setPageVariableIfNotSetBefore(variableName, md.renderInline($(node).html()));
+      }
+    });
+
+    this.variableLookup[fileName] = pageVariables;
+
+    return { ...importedVariables, ...pageVariables };
+  }
+
+  /**
+   * Extracts inner variables of the content, resolving new imports with respect to
+   * the supplied contentFilePath.
+   * @param content to extract from
+   * @param contentFilePath of the content
+   * @return {[]} Array of files that were imported
+   */
+  extractInnerVariables(content, contentFilePath) {
```

Comment:
- reduction of a parameter ( `config` ): due to site variable processing being centered in here now
- `context` -> `contentFilePath` ( let's not pass in what's more than necessary )